### DISCUSSION TITLE
SSL设置时出现错误

### DISCUSSION THREAD

**Discussion by fhfzfy1231** [Jinnrry/PMail#203] [Q&A] [answered]
前面的步骤都没有问题，SSL部署时选择的是HTTP请求，已知转圈，无法完成，看软件命令行显示如下内容
[error][2024-09-16 14:06:52][/home/runner/work/PMail/PMail/server/utils/async/async.go:68]goroutine panic:error: one or more domains had a problem:
[pmail.foxlane.club] acme: error: 400 :: urn:ietf:params:acme:error:connection :: 112.83.94.42: Fetching http://pmail.foxlane.club/.well-known/acme-challenge/vIXU3rzeJIE2JTzmHlCGTYWJjCwH6_P-2J91AAFKP10: Timeout during connect (likely firewall problem)
[pop.foxlane.club] acme: error: 400 :: urn:ietf:params:acme:error:connection :: 112.83.94.42: Fetching http://pop.foxlane.club/.well-known/acme-challenge/tHApqZfoFTVgvPf7-3qDFqgiEusmMjvvIyFpietha0A: Timeout during connect (likely firewall problem)
[smtp.foxlane.club] acme: error: 400 :: urn:ietf:params:acme:error:connection :: 112.83.94.42: Fetching http://smtp.foxlane.club/.well-known/acme-challenge/mC2QiYoyODhdx0KQy2SSoxIdj1Izza-EUILyOvv4Qzk: Timeout during connect (likely firewall problem)

 goroutine 116 [running]:
runtime/debug.Stack()
        /opt/hostedtoolcache/go/1.22.6/x64/src/runtime/debug/stack.+0x5e
github.com/Jinnrry/pmail/utils/async.(*Async).HandleErrRecover(0xc00013a180, {0xaf4000, 0xc000364870})
        /home/runner/work/PMail/PMail/server/utils/async/async.go:68 +0xd2
github.com/Jinnrry/pmail/utils/async.(*Async).Process.func1.1()
        /home/runner/work/PMail/PMail/server/utils/async/async.go:45 +0x3a
panic({0xaf4000?, 0xc000364870?})
        /opt/hostedtoolcache/go/1.22.6/x64/src/runtime/panic.go:770 +0x132
github.com/Jinnrry/pmail/services/setup/ssl.generateCertificate.func1({0x0?, 0xc000421fd0?})
        /home/runner/work/PMail/PMail/server/services/setup/ssl/ssl.go:201 +0x1bd
github.com/Jinnrry/pmail/utils/async.(*Async).Process.func1()
        /home/runner/work/PMail/PMail/server/utils/async/async.go:48 +0x58
created by github.com/Jinnrry/pmail/utils/async.(*Async).Process in goroutine 65
        /home/runner/work/PMail/PMail/server/utils/async/async.go:42 +0x8f

除此以外我也尝试过使用DNS解析，但是把它生成的内容放到DNSpod后也是一直转圈，而手动DNS我不知道是解析哪个ip，是pamil还是pop还是smtp

**Comment by Jinnrry** [comment, marked as answer]
1、自动dns的问题`Timeout during connect (likely firewall problem)` 似乎是你服务器外网访问不通？你是不是国内的服务器，没有备案，被服务商阻断了？
2、手动dns需要匹配你配置的所有域名，使用泛域名，或者多域名证书
